File: Basic/Maximum Occuring Character/maximum-occuring-character.py
#User function Template for python3

class Solution:
    
    #Function to find the maximum occurring character in a string.
    def getMaxOccurringChar(self,s):
        f = {}
        maxi = 0
        val = ''
        for i in s:
            if i not in f:
                f[i] = 1
            else: 
                f[i] +=1
        for i in f:
            if f[i]>maxi:
                maxi = f[i]
                val = i
            if f[i]==maxi:
                if val>i:
                    val=i
        return val


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import atexit
import io
import sys

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER

@atexit.register

def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())

if __name__=='__main__':
    t = int(input())
    for i in range(t):
        s=str(input())
        print(Solution().getMaxOccurringChar(s))
# } Driver Code Ends

File: Basic/Maximum Occuring Character/maximum-occuring-character.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends


class Solution
{
    public:
    //Function to find the maximum occurring character in a string.
    char getMaxOccuringChar(string str)
    {
        // Your code here
        int a[26]={0};
        
        int n=str.length();
        for(int i=0; i<n; i++){
            if(str[i]>='a' && str[i]<='z'){
                int x = str[i]-'a';
                a[x]++;
            }
            if(str[i]>='A' && str[i]<='Z'){
                int y = str[i]-'A';
                a[y]++;
            }
        }
        int maxi = 0;
        for(int i=0; i<26; i++){
            //std::cout<<a[i]<<std::endl;
            if(a[i]>a[maxi]){
                maxi = i;
            }
        }
        return('a'+maxi);
    }

};

//{ Driver Code Starts.

int main()
{
   
    int t;
    cin >> t;
    while(t--)
    {
        string str;
        cin >> str;
    	Solution obj;
        cout<< obj.getMaxOccuringChar(str)<<endl;
    }
}
// } Driver Code Ends